### PR TITLE
Generalize logging in scheduled jobs

### PR DESCRIPTION
**Pull request by rkondratowicz**
Now that app insights alerts for jobs rely on certain entries in logs, it is important to keep the logs consistent and not change them (unless alert is updated as well).

Hopefully after this change this will be less likely to happen in the future.

Note: not using inheritance intentionally.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/utils/LoggingHelper.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.slf4j.Logger;
+
+public final class LoggingHelper {
+
+    private LoggingHelper() {
+        // util class
+    }
+
+    public static void wrapWithJobLog(Logger logger, String taskName, Runnable runnable) {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/utils/LoggingHelper.java]: I'm open to suggestions on alternative name :)

File: src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/utils/LoggingHelper.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package uk.gov.hmcts.reform.blobrouter.tasks.utils;
+
+import org.slf4j.Logger;
+
+public final class LoggingHelper {
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/utils/LoggingHelper.java]: this is also executing the tasks ? so probably `ScheduledTaskRunner` ?

File: src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/utils/LoggingHelper.java (hunk 1 above)
Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/utils/LoggingHelper.java]: `executeWithJobLog` ?

File: src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/utils/LoggingHelper.java (hunk 2 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/utils/LoggingHelper.java]: it's not running scheduled tasks.
Scheduled tasks are run automagically by Spring, then scheduled tasks use this class to surround code with logs.

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/utils/LoggingHelper.java]: Or maybe `JobRunner`?

File: src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/utils/LoggingHelper.java (hunk 1 above)
Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/utils/LoggingHelper.java]: I'd be even fine with `run`.

File: src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/utils/LoggingHelper.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+    private LoggingHelper() {
+        // util class
+    }
+
+    public static void wrapWithJobLog(Logger logger, String taskName, Runnable runnable) {
+        logger.info("Started {} job", taskName); // DO NOT change this message. App insights alerts are based on it.
+        runnable.run();
+        logger.info("Finished {} job", taskName);
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/utils/LoggingHelper.java]: Not sure, but maybe it's worth handling `Exception` here.

File: src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/utils/LoggingHelper.java (hunk 2 above)
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/utils/LoggingHelper.java]: perhaps worth considering continuously developing/improving/bringing-up-to-speed app insights setup for the service by making #204 pass. Currently it's reporting that rejected tasks are not working properly. Same thing is also done in send letter service too. Then logging will be more trivial than ever. and will not involve in tampering schedules themselves

btw, @nitinprabhu, what you are referring perhaps closely looks like the #102, which is, again, setup for #204 thing to happen

Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/utils/LoggingHelper.java]: `JobRunner` sounds good @lgonczar  @rkondratowicz

**Pull request closed without merging**